optimum/tpu/modeling.py

```diff
@@ -20,11 +20,14 @@
 
 from loguru import logger
 from transformers import AutoModelForCausalLM as BaseAutoModelForCausalLM
-from transformers.utils import is_accelerate_available
 
+from optimum.tpu.modeling_gemma import TpuGemmaForCausalLM
+
+def config_name_to_class(pretrained_model_name_or_path: str):
+    if "gemma" in pretrained_model_name_or_path:
```

Comment:
Relying on `config.model_type` is more robust as a Gemma checkpoint will not necessarily have "gemma" in its name:
```suggestion
    config = AutoConfig.from_pretrained(pretrained_model_name_or_path)
    if config.model_type == "gemma":
```